src/couch_dist/src/couch_dist.erl

```diff
@@ -0,0 +1,87 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(couch_dist).
+
+-export([childspecs/0, listen/2, accept/1, accept_connection/5,
+    setup/5, close/1, select/1, is_node_name/1]).
+
+childspecs() ->
+    {ok, [{ssl_dist_sup, {ssl_dist_sup, start_link, []},
+        permanent, infinity, supervisor, [ssl_dist_sup]}]}.
+
+listen(Name, Host) ->
+    inet_tls_dist:listen(Name, Host).
+
+
+accept(Listen) ->
+    inet_tls_dist:accept(Listen).
```

Comment:
I wonder if this would work with `no_tls = true` when TCP is used?

Noticed that in `inet_tls_dist` it calls `ssl:handshake/3` in accept logic https://github.com/erlang/otp/blob/master/lib/ssl/src/inet_tls_dist.erl#L285.